Formula/ignition-utils0.rb

```diff
@@ -0,0 +1,60 @@
+class IgnitionUtils0 < Formula
+  desc "General purpose classes and functions designed for robotic applications"
+  homepage "https://github.com/ignitionrobotics/ign-utils"
+  url "https://github.com/ignitionrobotics/ign-utils/archive/20a2b6fc0030cd3abbac36d476da8e36d6794404.tar.gz"
+  version "0.1.0~pre0~0~20201223~20a2b6f"
+  sha256 "fa5584d13116ef2b00a1776a8acacdc27f1d5755e416763bb836bb0490558bfd"
+  license "Apache-2.0"
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "pkg-config" => [:build, :test]
+  depends_on "ignition-cmake2"
+
+  def install
+    cmake_args = std_cmake_args
+    cmake_args << "-DBUILD_TESTING=Off"
+    system "cmake", ".", *cmake_args
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.cpp").write <<-EOS
+      #include <ignition/utils/ImplPtr.hh>
+      class SomeClassPrivate
+      {
+      };
+      class SomeClass
```

Comment:
I think `dataPtr` has to be constructed using `MakeImpl`
```suggestion
      class SomeClass: dataPtr(ignition::utils::MakeImpl<SomeClassPrivate>())
```

Reply:
that's the class definition, not the constructor

* [![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=generic-release-homebrew_triggered_bottle_builder%2Flabel%3Dosx_mojave&build=189)](https://build.osrfoundation.org/job/generic-release-homebrew_triggered_bottle_builder/label=osx_mojave/189/) https://build.osrfoundation.org/job/generic-release-homebrew_triggered_bottle_builder/label=osx_mojave/189/console

Reply:
Oops, sorry, it should have been
```c++
class SomeClass
{
  SomeClass(): dataPtr(ignition::utils::MakeImpl<SomeClassPrivate>()) 
  {}
...
```

Reply:
trying member initializer in d552c9b

Reply:
> Oops, sorry, it should have been

sorry I pushed before I saw your comment. Do you think it's better to define the default constructor or use an inline member initializer?

Reply:
The most common use case is going to be via the default constructor as shown in https://github.com/osrf/sdformat/pull/447/files#diff-a6e2168710bc843df1acd00632454dbe90626aa4f0e21cef75c9c0027d15a482R42, but I don't think it matters much for this test.

Reply:
:+1: